### PR TITLE
Do not configure TLS certificates when ACME is enabled.

### PR DESCRIPTION
**Pull request by guylabs**
Hi

First thanks a lot for maintaining this Rancher catalog item! 

After the latest upgrade I have the issue that when custom TLS certificates and ACME are configured, the HTTPS endpoint is always configured with the custom TLS certificate instead of the ACME certificates.

Therefore the ACME certificates are never taken and it looks like after the latest changes with the default SSL certificates written to `` it will always configure them when you enable HTTPS.

Nothing is written about this state in the main ACME documentation of Traefik (https://docs.traefik.io/configuration/acme/) nor in the entrypoints documentation, so maybe the default is to take the custom certificates first before the ACME.

This pull request fixes this issue by checking if ACME is enabled or not and depending on that it will write the proper configuration.

Would be nice if this would be part of the next release.

Regards

Guy

### CODE REVIEW

**Comment by rawmind0** [comment]
Hi @guylabs, thanks so much for the nice PR. :)

Could you please solve conflict in file `root/opt/traefik/bin/traefik.toml.sh` ??

Thanks....

**Comment by guylabs** [comment]
Hi @rawmind0 

done.

File: root/opt/traefik/bin/traefik.toml.sh [hunk 1]
```diff
@@ -146,25 +146,30 @@ x8NPFRMB44AWyLI=
 echo "${TRAEFIK_SSL_KEY}" > ${TRAEFIK_SSL_KEY_FILE}
 echo "${TRAEFIK_SSL_CRT}" > ${TRAEFIK_SSL_CRT_FILE}
 
-filelist=`ls -1 ${TRAEFIK_SSL_PATH}/*.key | rev | cut -d"." -f2- | rev`
-RC=`echo $?`
 
-if [ $RC -eq 0 ]; then
-    TRAEFIK_ENTRYPOINTS_HTTPS="\
+if [ "X${TRAEFIK_HTTPS_ENABLE}" == "Xtrue" ] || [ "X${TRAEFIK_HTTPS_ENABLE}" == "Xonly" ] || [ "X${TRAEFIK_ACME_ENABLE}" == "Xtrue" ]; then
```
Comment (rawmind0) [inline root/opt/traefik/bin/traefik.toml.sh]: Don't need to add last OR in the if command.. To enable ACME support it's mandatory enable HTTPS.
```
-if [ "X${TRAEFIK_HTTPS_ENABLE}" == "Xtrue" ] || [ "X${TRAEFIK_HTTPS_ENABLE}" == "Xonly" ] || [ "X${TRAEFIK_ACME_ENABLE}" == "Xtrue" ]; then
+if [ "X${TRAEFIK_HTTPS_ENABLE}" == "Xtrue" ] || [ "X${TRAEFIK_HTTPS_ENABLE}" == "Xonly" ]; then
```

Comment (guylabs) [reply root/opt/traefik/bin/traefik.toml.sh]: Done

**Comment by rawmind0** [comment]
Thanks @guylabs ...

LGTM...

**Comment by rawmind0** [comment]
Included at release rawmind/alpine-traefik:1.4.5-3

**Comment by rayout** [comment]
For us this is a very poor functionality.
We have a purchased certificate for one domain. And earlier we could use it simultaneously with ACME for others domains. Now it does not work.

**Pull request merged**